mobile/library/common/http/client.cc

```diff
@@ -659,9 +659,9 @@ Client::DirectStreamSharedPtr Client::getStream(envoy_stream_t stream,
   if (direct_stream_pair_it != streams_.end()) {
     return direct_stream_pair_it->second;
   }
-  if (direct_stream_pair_it == streams_.end() && get_stream_filters == ALLOW_FOR_ALL_STREAMS) {
-    direct_stream_pair_it = closed_streams_.find(stream);
-    if (direct_stream_pair_it != closed_streams_.end()) {
+  if (get_stream_filters == ALLOW_FOR_ALL_STREAMS) {
+    if (direct_stream_pair_it = closed_streams_.find(stream);
```

Comment:
You can have assignment in an if, followed by another statement that does a boolean check, but I agree it can be a confusing syntax, so I changed it to your suggestion.  What I did anyway is only really useful if you want to reduce the scope of a variable declaration, but we're not declaring any new variable in this, so your suggestion is cleaner.  Done.